Mathlib/Algebra/Group/Units.lean

```diff
@@ -639,6 +639,12 @@ def IsUnit [Monoid M] (a : M) : Prop :=
 #align is_unit IsUnit
 #align is_add_unit IsAddUnit
 
+@[to_additive]
+lemma isUnit_iff_exists [Monoid M] {x : M} : IsUnit x ↔ ∃ b, x * b = 1 ∧ b * x = 1 := by
+  refine ⟨fun ⟨u, hu⟩ => ?_, fun ⟨b, h1b, h2b⟩ => ⟨⟨x, b, h1b, h2b⟩, rfl⟩⟩
+  subst x
+  exact ⟨u.inv, u.val_inv, u.inv_val⟩
```

Comment:
Is this the same lemma as the one in #10719 (not merged, yet)?

Reply:
Not, exactly (one existential vs two), but I might be happy with that one as well.

Reply:
Ah, of course!  I am happy with having both lemmas, although ideally they should end up next to one another and, even better, have doc-strings mentioning each other.  This may simply be the job of a PR following the merging of the two existing PRs.